aten/src/ATen/native/cpu/layer_norm_kernel.cpp

```diff
@@ -34,28 +34,34 @@ void LayerNormKernelImplInternal(
   const T c = T(1) / static_cast<T>(N);
   const bool gamma_null = gamma_data == nullptr;
   const bool beta_null = beta_data == nullptr;
-  for (int64_t i = 0; i < M; ++i) {
-    const T* X_ptr = X_data + i * N;
-    T* Y_ptr = Y_data + i * N;
-    T mean_val = T(0);
-    T rstd_val = T(0);
-    for (int64_t j = 0; j < N; ++j) {
-      mean_val += X_ptr[j];
-      rstd_val += X_ptr[j] * X_ptr[j];
-    }
-    mean_val *= c;
-    rstd_val = std::max(rstd_val * c - mean_val * mean_val, T(0));
-    rstd_val = T(1) / std::sqrt(rstd_val + eps);
-    const T scale = rstd_val;
-    const T bias = -rstd_val * mean_val;
-    for (int64_t j = 0; j < N; ++j) {
-      const T gamma_v = gamma_null ? T(1) : gamma_data[j];
-      const T beta_v = beta_null ? T(0) : beta_data[j];
-      Y_ptr[j] = (X_ptr[j] * scale + bias) * gamma_v + beta_v;
+  int num_tasks = at::get_num_threads();
+  at::parallel_for(0, num_tasks, 1, [&](int64_t start, int64_t end) {
+    const int64_t M_per_thread = (M + num_tasks - 1) / num_tasks;
+    const int64_t M_start = std::min(start * M_per_thread, M);
+    const int64_t M_end = std::min(end * M_per_thread, M);
```

Comment:
this is not how `at::parallel_for` is typically used, you don't need to compute the number of tasks and start and end by yourself - check https://github.com/pytorch/pytorch/pull/19105 for example,i

you typically just need to write, e.g.
```
at::parallel_for(0, M, <choose grain size>, [&](int64_t start, int64_t end) { 
for (int64_t i = start; i < end; ++i) { ... }
});
```

Reply:
Thanks! Updated the PR. One thing to double check: When I set the `<choose grain size>` to 1, does this mean the loop is iterated in an interleaving way, or in a block-cyclic way?

Reply:
As @ilia-cher told me, the loop is iterated in a "block-cyclic" way: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/ParallelNative.h#L20